Source/AutoCloseLetters/AutoCloseLetters.cs

```diff
@@ -33,12 +33,12 @@ public static class AutoCloseLetters_Patch
         [HarmonyPrefix]
         public static void DeleteLettersInstantly()
         {
-            Find.LetterStack.LettersListForReading.Where(x => x.arrivalTick < Find.TickManager.TicksGame - (60000 * Prefs.AutosaveIntervalDays)).ToList().ForEach(Remove);
+            Find.LetterStack.LettersListForReading.Where(x => x.arrivalTick < Find.TickManager.TicksGame - (60000 * Prefs.AutosaveIntervalDays) && x.disappearAtTick < 0).ToList().ForEach(Remove);
```

Comment:
Looks like you're assuming `x` is LetterWithTimeout to access the `disappearAtTick` property, when it is Letter. You could replace the check for the property with just ` && !(x is LetterWithTimeout)`

```suggestion
            Find.LetterStack.LettersListForReading.Where(x => x.arrivalTick < Find.TickManager.TicksGame - (60000 * Prefs.AutosaveIntervalDays) && !(x is LetterWithTimeout)).ToList().ForEach(Remove);
```